### DISCUSSION TITLE
Allow blocking flows to return messages displayed to the user

### DISCUSSION THREAD

**Discussion by max-mayorov** [directus/directus#20355] [Feature Requests]
### Summary

Currently, when a blocking flow is ended on the unhappy path, the only message that user sees is 'An unexpected error occured'. However it would be nice to let flow return a meaningful message that will be displayed to the user.

New exception class shoould be added that will allow error message to propagate to the user level.

### Motivation

For example, we created a flow to enforce certain user permissions. E.g. some combination of field values is not allowed. We can block such combinations with a flow, but we cannot inform the user why the record cannot be updated. We would like instead show user a message 'This combination is not allowed'.

### Detailed Design

[This piece of code](https://github.com/directus/directus/blob/6509e06a5454befe1ed9cdbd53ba86554b357a35/api/src/middleware/error-handler.ts#L54C6-L54C6) always return 'Unexpected error' when user is not admin.

Add new exception class that will allow error message to propagate down to the user level.
Something like that in script block:
```
const condition = /* condition check */
if(condition) {
  throw new BlockedError('Condition didn't satisfied, update is blocked');
}
```
The user should see the message 'Condition didn't satisfied, update is blocked' instead of 'Unexpected error'

### Must Have:
- Return user-friendly error messages from blocking flows that are displayed to the user.

### Drawbacks

none

### Alternatives

none

### Adoption Strategy

new feature, fully compatible with previous versions

**Comment by lpkobamn** [comment]
absolutely need this!

**Comment by ched-dev** [comment]
I opened an issue for this because I think it's more of a bug than a feature request: https://github.com/directus/directus/issues/19612
